node/api/client/host.go

```diff
@@ -1,24 +1,84 @@
 package client
 
 import (
+	"fmt"
 	"net/url"
 	"strconv"
 
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/node/api"
 )
 
+// HostParam is a parameter in the host's settings that can be changed via the
+// API. It is primarily used as a helper struct to ensure type safety.
+type HostParam string
```

Comment:
this makes me a bit uneasy. Probably because the HTTP API doesn't match the Go API: the HTTP API takes a key+value pair, whereas the Go API takes a full `HostInternalSettings` struct.

I prefer the latter form because you can make multiple changes atomically. But the former form more closely matches how people will actually change their settings: one at a time.

I think my preference would be to update the `/host` route to accept a full `HostInternalSettings` object. But for now, a big list of `consts` is probably the best we can do. (btw, they can be `const`, not `var`.)

Reply:
well, we could make the change in a backwards-compatible way, and have the `client` package only support the new way. Something to think about for later.